src/Result/Failure.php

```diff
@@ -56,11 +76,19 @@ public static function fromCheckAndReason(CheckInterface $check, $reason)
     /**
      * @param string $name
      * @param CodeExecutionException $e
+     * @param string|null $expectedOutput
+     * @param string|null $actualOutput
+     * @param string|null $errors
      * @return static
      */
-    public static function fromNameAndCodeExecutionFailure($name, CodeExecutionException $e)
-    {
-        return new static($name, $e->getMessage());
+    public static function fromNameAndCodeExecutionFailure(
+        $name,
+        CodeExecutionException $e,
+        $expectedOutput = null,
+        $actualOutput = null,
```

Comment:
It seems `$expectedOutput` & `$errors` come from the exception anyway? So could we not reduce the arguments to this constructor and grab them from the exception in this method?